api/v1/etcd_types.go

```diff
@@ -253,13 +233,14 @@ type LastOperation struct {
 type Status struct {
 	Etcd            CrossVersionObjectReference `json:"etcd,omitempty"`
 	Conditions      []Condition                 `json:"conditions,omitempty"`
-	CurrentReplicas int32                       `json:"currentReplicas,omitempty"`
+	CurrentReplicas *int32                      `json:"currentReplicas,omitempty"`
 	Endpoints       []corev1.Endpoints          `json:"endpoints,omitempty"`
-	LastError       string                      `json:"lastError,omitempty"`
-	Replicas        int32                       `json:"replicas,omitempty"`
-	ReadyReplicas   int32                       `json:"readyReplicas,omitempty"`
-	Ready           bool                        `json:"ready,omitempty"`
-	UpdatedReplicas int32                       `json:"updatedReplicas,omitempty"`
+	ServiceName     *string                     `json:"serviceName,omitempty"`
+	LastError       *string                     `json:"lastError,omitempty"`
+	Replicas        *int32                      `json:"replicas,omitempty"`
+	ReadyReplicas   *int32                      `json:"readyReplicas,omitempty"`
+	Ready           *bool                       `json:"ready,omitempty"`
+	UpdatedReplicas *int32                      `json:"updatedReplicas,omitempty"`
```

Comment:
We will need a `LabelSelector` field in the `Status` if we want to implement [`Scale` sub-resource](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#scale-subresource), which we should.

Reply:
It is optional but the default behaviour if it is not supplied is quite useless. So, might be good to have it.

> If there is no value under the LabelSelectorPath in the custom resource, the status selector value in the /scale subresource will default to the empty string.

Also,
> It must be set to work with HPA.

Though we don't want to use HPA with our CRD but you never know which other component working with the `Scale` subresource expects a proper value here.